Cleanup config-nodejs in preparation for Single Executable Application for device-agent

alexbigkid opened the pull request:
- moved test files into tests directory
- updated to use eslint
- fixed all lint errors
- needed to fix nodejs 20 related unit tests problems
- had to disable windows testing on the pipeline, because the functionality tested is not windows compatible
- created github action pipeline to lint, unit test, build and publish package to npmjs.com

ericvanbuhler commented:
@alexbigkid I'm surprised about the Windows issue since this package is used in the CLI on Windows. What was the specific problem?

ericvanbuhler commented on .github/workflows/pipeline.yml:
```diff
@@ -0,0 +1,76 @@
+name: alwaysai/codecs
```
This should be `alwaysai/config-nodejs`, right?

alexbigkid replied:
@ericvanbuhler Yes! Thank you! Copy and paste error

alexbigkid commented:
@ericvanbuhler, I was also surprised, that unit tests were failing on windows. Here is the link to the run of unit tests which were failing: https://github.com/alwaysai/config-nodejs/actions/runs/8906683766/job/24459230985
I shortly looked into why it is failing. The unit tests call function which do not sure exist on windows. At least I am not sure how those calls gets translated on Windows. In file config-file-schema.test.ts, the tests: "write throws EACCES", there is a call `chmodSync`. So may be the production code is fine, but the tests seems not to work on Windows. Fixing them would be out of scope for this ticket. But we can create one to look into later.

alexbigkid commented:
@ericvanbuhler : I updated eslint. And the new eslint flagged few issues with prod code, which I fixed. They were minor, but were producing lint errors, causing pipeline to fail. The one I was not able to fix: I disabled the eslint error. But created a ticket to look at later. Ticket EI-1252.

ericvanbuhler commented:
> @ericvanbuhler, I was also surprised, that unit tests were failing on windows. Here is the link to the run of unit tests which were failing: https://github.com/alwaysai/config-nodejs/actions/runs/8906683766/job/24459230985 I shortly looked into why it is failing. The unit tests call function which do not sure exist on windows. At least I am not sure how those calls gets translated on Windows. In file config-file-schema.test.ts, the tests: "write throws EACCES", there is a call `chmodSync`. So may be the production code is fine, but the tests seems not to work on Windows. Fixing them would be out of scope for this ticket. But we can create one to look into later.

I see, so it is the test that is not supported. I imagine those tests were written assuming they would always be on Llinux/Mac so that makes sense.
